### ISSUE TITLE
[BE] Don't bundle tests in Python Lambda packages

### ISSUE THREAD

**Issue by austinbyers** [panther-labs/panther#2669]
# Background
The Python Lambda functions (`panther-policy-engine`, `panther-rules-engine`) include both the `src` and the `tests` directories in their zipfiles.

It's not a big deal, but it means unnecessary code has to be packaged in every deployment. More importantly, the test code is visible to all users (including enterprise) and could in theory be imported from the source code. It's possible a weird test could affect production if it was accidentally loaded

In other words, there's no good reason to include the tests and some minor potential problems because we do. Let's remove them, keep it simple

# Acceptance Criteria

- Python Lambda deployment packages contain only source files, not their tests

# Implementation Details

Either:

- Move the zip root target directory to `src` (which will probably break the imports)
- Explicitly skip the tests during `mage deploy / publish` operations
